lib/src/cache/cache.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'package:w_common/disposable.dart';
+import 'package:w_common/func.dart';
+
+/// Immutable payload that indicates a change in a [Cache].
+class CacheContext<TIdentifier, TValue> {
```

Comment:
`CachePayload`? `CachedItem`? `CacheResident`? Naming is hard. I don't love "context".